### PR TITLE
Add sort command 

### PR DESCRIPTION
**Pull request by sunzihan23**

### CODE REVIEW

File: src/main/java/seedu/address/model/course/Course.java [hunk 1]
```diff
@@ -130,6 +134,23 @@ public void updateFilteredStudentList(Predicate<Student> predicate) {
...
+        sortedStudents.setComparator(comparator);
+        this.filteredStudents = new FilteredList<>(sortedStudents);
+    }
+
+    /**
+     * Returns an unmodifiable view of the filtered list of {@code Student} in the original order
+     */
+    public void resetFilteredStudentList() {
+        this.filteredStudents = new FilteredList<>(this.students.asUnmodifiableObservableList());
+    }
```
Comment (gongg21) [inline src/main/java/seedu/address/model/course/Course.java:152]: Nice and smart implementation of resetting the filtered student list.

File: src/main/java/seedu/address/logic/commands/SortCommand.java [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+
+        StageManager stageManager = StageManager.getCurrent();
+        Course course = stageManager.getCurrentCourse();
+
+        // Custom comparator to sort students by their student ranks
+        Comparator<Student> tagComparator = Comparator.comparing(student -> student.getTag().ranking);
+
+        course.updateSortedStudentList(tagComparator);
+
+        return new CommandResult(MESSAGE_SUCCESS);
```
Comment (gongg21) [inline src/main/java/seedu/address/logic/commands/SortCommand.java:39]: LGTM

**Review by gongg21 — COMMENTED** [review]
Good implementation of Sort command and improvements to List and Find.

File: src/main/java/seedu/address/logic/commands/SortCommand.java (hunk 2 above)
Comment (devanshubisht) [reply src/main/java/seedu/address/logic/commands/SortCommand.java:39]: Good use of comment to make it clear

**Pull request merged**